docs/src/components/TableCode.tsx

```diff
@@ -18,23 +18,30 @@ export function TableCode({ children, printWidth = 40, ...props }: TableCodeProp
     let groupIndex = 0;
     parts.forEach((child, index) => {
       if (index === 0) {
-        unionGroups.push([child]);
-        return;
+        unionGroups.push([]);
       }
 
-      if (getChildrenText(child).trim() === '|' && depth < 1) {
-        groupIndex += 1;
+      const str = getChildrenText(child);
+
+      // Solo function return values shouldn't be broken up, e.g. in:
+      // `(value) => string | string[] | null | Promise`
+      if (str.includes('=>') && depth < 1) {
+        depth += 1;
+      }
+
+      if (str.trim() === '|' && depth < 1 && index !== 0) {
         unionGroups.push([]);
+        groupIndex += 1;
         return;
       }
 
-      if (getChildrenText(child).trim() === '(') {
+      str.split('(').forEach(() => {
         depth += 1;
-      }
+      });
 
-      if (getChildrenText(child).trim() === ')') {
+      str.split(')').forEach(() => {
         depth -= 1;
-      }
+      });
```

Comment:
before (incorrect, `anchor` function return value members are on the same level as the function itself)

<img width="808" alt="image" src="https://github.com/user-attachments/assets/472bbd84-2fc7-4fda-a2a5-3445fc88967b" />


after (correct, `anchor` is a union type, and one of its members is a function that returns a union type):

<img width="813" alt="image" src="https://github.com/user-attachments/assets/ef1b3efe-6b7e-4388-b4c8-a7c008c64d53" />